lldb/source/Target/Target.cpp

```diff
@@ -4557,7 +4557,17 @@ const char *TargetProperties::GetDisassemblyCPU() const {
   const uint32_t idx = ePropertyDisassemblyCPU;
   llvm::StringRef str = GetPropertyAtIndexAs<llvm::StringRef>(
       idx, g_target_properties[idx].default_cstr_value);
-  return str.empty() ? nullptr : str.data();
+
+  // If the user has an explict setting, then use that.
+  if (!str.empty())
+    return str.data();
+
+  // Next, check if the architecture has a default CPU setting.
+  std::string cpu = m_target->GetArchitecture().GetClangTargetCPU();
+  if (!cpu.empty())
+    return ConstString(cpu).GetCString();
+
+  return nullptr;
```

Comment:
I wouldn't put this into this function as we want the `GetDisassemblyCPU()` call to always get the properties from the properties only without any additional work. I don't think any other properties accessors do extra work like this. Maybe we should move this blurb of code inside a new `lldb_private::Target` method:
```
llvm::StringRef Target::GetDisassemblyCPUName() const {
  // Try to get the GPU name from the target settings first.
  llvm::StringRef cpu(GetDisassemblyCPU());
  if (!cpu.empty())
    return cpu;
  // Next, check if the architecture has a default CPU setting.
  std::string cpu = m_target->GetArchitecture().GetClangTargetCPU();
  if (!cpu.empty())
    return ConstString(cpu).GetCString();

  return nullptr;
}
```

Reply:
@clayborg  I reverted the change to `GetDisassemblyCPU()` and modified the Target code to update the property when the target architecture is changed: [ede48b1](https://github.com/clayborg/llvm-project/pull/24/commits/ede48b1df0341b96c5155a4f03d44a527e3baf28).

Let me know what you think.